ReactWindows/ReactNative/Views/Image/ReactImageManager.cs

```diff
@@ -107,41 +111,51 @@ public void SetResizeMode(Border view, string resizeMode)
         /// Set the source URI of the image.
         /// </summary>
         /// <param name="view">The image view instance.</param>
-        /// <param name="source">The source URI.</param>
+        /// <param name="sources">The source URI.</param>
         [ReactProp("src")]
-        public void SetSource(Border view, string source)
+        public void SetSource(Border view, JArray sources)
         {
-            var imageBrush = (ImageBrush)view.Background;
+            // There is no image source
+            if (sources.Count == 0)
+            {
+                throw new ArgumentException("Sources must not be empty.", nameof(sources));
+            }
 
-            view.GetReactContext()
-                .GetNativeModule<UIManagerModule>()
-                .EventDispatcher
-                .DispatchEvent(
-                    new ReactImageLoadEvent(
-                        view.GetTag(),
-                        ReactImageLoadEvent.OnLoadStart));
+            // Optimize for the case where we have just one uri, case in which we don't need the sizes
+            if (sources.Count == 1)
+            {
+                var uri = ((JObject)sources[0]).Value<string>("uri");
+                SetUriFromSingleSource(view, uri);
+            }
+            else
+            {
+                var viewSources = default(Dictionary<string, double>);
+                var tag = view.GetTag();
+                if (_imageSources.TryGetValue(tag, out viewSources))
+                {
+                    viewSources.Clear();
+                }
+                else
+                {
+                    viewSources = new Dictionary<string, double>();
+                }
 
-            var reference = _imageCache.Get(source);
-            var subscription = reference.LoadedObservable.Subscribe(
-                _ =>
+                foreach (var source in sources)
                 {
-                    imageBrush.ImageSource = reference.Image;
-                    OnImageOpened(view);
-                },
-                _ => OnImageFailed(view));
+                    var sourceData = (JObject)source;
+                    viewSources.Add(sourceData.Value<string>("uri"), sourceData.Value<double>("width") * sourceData.Value<double>("height"));
+                }
 
-            var disposable = default(SerialDisposable);
-            if (!_disposables.TryGetValue(view, out disposable))
-            {
-                disposable = new SerialDisposable();
-                _disposables.Add(view, disposable);
-            }
+                _imageSources.Add(tag, viewSources);
```

Comment:
Sorry, instead of using the mutable setter, can you just move the `_imageSources.Add(tag, viewSources);` line into the else block above?

---
In reply to: [76254950](https://github.com/ReactWindows/react-native-windows/pull/625#discussion_r76254950) [](ancestors = 76254950)